docs/learn/learn-staking-miner.md

```diff
@@ -0,0 +1,122 @@
+The best solution in each round is rewarded. All correct solutions will get their bond back and the
+ones that submit invalid solutions will lose their bond.
+
+## NPoS election optimization
+
+A basic election solution is a simple distribution of stake across validators, but this can be
+optimized for better distribution equaling a higher security score. The staking miner does not act
+as a validator and focuses solely on the election result and optimization of the solution. It
+connects to a specified chain and keeps listening to new signed phase of the election pallet in
+order to submit solutions to the NPoS election. When the correct time comes, it computes its
+solution and submits it to the chain. The default miner algorithm is sequential Phragmén with a
+configurable number of balancing iterations that improve the score.
+
+![NPoS election optimization](../assets/staking-miner/NPoS-election-optimization.png)
+
+## Signed Phase of the election pallet
+
+The election provider pallet `pallet_election_provider_multi_phase` is divided into two phases,
+**signed** and **unsigned**. At the end of the pallet's timeline, the function `elect()` is called.
+
+```
+                                                                   elect()
+                +   <--T::SignedPhase-->  +  <--T::UnsignedPhase-->   +
+  +-------------------------------------------------------------------+
+   Phase::Off   +       Phase::Signed     +      Phase::Unsigned      +
+```
+
+Solutions provided by the staking miner can only be submitted during the signed phase. Solutions are
+submitted and queued on the chain as a `RawSolution`. Once submitted, a solution cannot be retracted
+by the originating account.
+
+`RawSolution` struct definition:
+
+```
+pub struct RawSolution<S> {
+    pub solution: S, // The solution itself
+    pub score: ElectionScore, // The claimed score of the solution.
+    pub round: u32, // The round at which this solution should be submitted.
+}
+```
+
+A maximum of `pallet::Config::MaxSignedSubmissions` will be stored on-chain and they will be sorted
+based on score. Higher the score the more optimal the election solution is. On both Polkadot and
+Kusama the
+['MaxSignedSubmissions'](https://github.com/paritytech/polkadot/blob/master/runtime/polkadot/src/lib.rs#L446)
+is set to {{ signed_max_submission }} submissions. This variable can be modified if needed through
+governance.
+
+Upon arrival of a new solution:
+
+1. If the queue is not full, it is stored in the appropriate sorted index.
+2. If the queue is full but the submitted solution is better than one of the queued ones, the worse
+   solution is discarded, the bond of the outgoing solution is returned, and the new solution is
+   stored in the correct index.
+3. If the queue is full and the solution is not an improvement compared to any of the queued ones,
+   it is instantly rejected and no bond is reserved.
```

Comment:
//cc @kianenigma mention that a solution with the "same score" will loose funds?